.github/workflows/ci_coverage.yml

```diff
@@ -0,0 +1,30 @@
+name: CI and Coverage
+on:
+  push:
+    paths-ignore:
+      - "**/*.md"
+      - "**/*.png"
+  pull_request:
+    paths-ignore:
+      - "**/*.md"
+      - "**/*.png"
+env:
+  GO_VERSION: 1.18.2
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+      - uses: actions/setup-go@v3
+        with:
+          go-version: ${{ env.GO_VERSION }}
+          cache: true
+      - name: Run coverage
+        run: go test ./... -race -coverprofile=coverage.xml -covermode=atomic
+      - name: Upload coverage to Codecov
+        run: |
+           curl -Os https://uploader.codecov.io/latest/linux/codecov
+           chmod +x codecov
+           ./codecov
```

Comment:
will this work w/o providing a token ?
https://about.codecov.io/blog/introducing-codecovs-new-uploader/

Reply:
> will this work w/o providing a token ? https://about.codecov.io/blog/introducing-codecovs-new-uploader/

yes, for public repositories, no token is needed.  
https://github.com/marketplace/actions/codecov